### PR TITLE
Fix empty URI for CosmosDBAccountData

### LINKED ISSUE #36104: [BUG] Deserializing CosmosDBAccountData throws a System.Uri.FormatException
### Library name and version

Azure.ResourceManager.CosmosDB 1.3.0

### Describe the bug

System.UriFormatException: Invalid URI: The URI is empty.
   at System.Uri.CreateThis(String uri, Boolean dontEscape, UriKind uriKind, UriCreationOptions& creationOptions)
   at System.Uri..ctor(String uriString)
   at Azure.ResourceManager.CosmosDB.CosmosDBAccountData.DeserializeCosmosDBAccountData(JsonElement element)
   at Azure.ResourceManager.CosmosDB.SubscriptionResourceExtensionClient.<GetCosmosDBAccountsAsync>b__22_1(JsonElement e)
   at Azure.Core.PageableHelpers.PageableImplementation`1.GetAsyncEnumerator(CancellationToken cancellationToken)+MoveNext()
   at Azure.Core.PageableHelpers.PageableImplementation`1.GetAsyncEnumerator(CancellationToken cancellationToken)+System.Threading.Tasks.Sources.IValueTaskSource<System.Boolean>.GetResult()

### Expected behavior

The only Uri field I see in the CosmosDBAccountData is KeyvaultUri which is Optional, an empty string should not throw an exception. The JsonElement in question is returned directly from the GetCosmosDBAccountAsync call

### Actual behavior

an empty string causes an exception when deserializing CosmosDBAccountData

### Reproduction Steps

                SubscriptionResource subscription = armClient.GetSubscriptionResource(SubscriptionResource.CreateResourceIdentifier(subscriptionId));

                AsyncPageable<CosmosDBAccountResource> cosmosDBAccounts = subscription.GetCosmosDBAccountsAsync(cancellationToken);

                try
                {
                    await foreach (CosmosDBAccountResource cosmosDBAccount in cosmosDBAccounts)
                    {
                        if (AccountIsInRegion(cosmosDBAccount, currentRegion))
                        {
                            // do some checks here using cosmosDBAccount Data
                        }
                    }
                }
                catch (Exception ex)
                {
                    _logger.LogError(ex, "error scanning for orphaned DB");
                }

### Environment

net 6.0 exe , Service Fabric app running on an Azure cloud test environment

### PR DESCRIPTION
**Pull request by live1206**
Resolves https://github.com/Azure/azure-sdk-for-net/issues/36104

# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/cosmosdb/Azure.ResourceManager.CosmosDB/tests/SessionRecords/RestorableDatabaseAccountTests/RestorableDatabaseAccountListAsync.json [hunk 1]
```diff
@@ -623,106 +623,107 @@
...
+                }
+            ],
+            "failoverPolicies": [
+                {
+                    "id": "r-database-account-6799-westus",
+                    "locationName": "West US",
+                    "failoverPriority": 0
+                }
+            ],
+            "keyVaultKeyUri": "",
```
Comment (live1206) [inline sdk/cosmosdb/Azure.ResourceManager.CosmosDB/tests/SessionRecords/RestorableDatabaseAccountTests/RestorableDatabaseAccountListAsync.json:695]: We can't intentionally create an instance of `URI` with empty string, by adding this line to service response to reproduce the issue.

File: sdk/cosmosdb/Azure.ResourceManager.CosmosDB/src/Generated/Models/CosmosDBAccountData.Serialization.cs [hunk 2]
```diff
@@ -552,7 +552,7 @@ internal static CosmosDBAccountData DeserializeCosmosDBAccountData(JsonElement e
                         }
                         if (property0.NameEquals("keyVaultKeyUri"u8))
                         {
-                            if (property0.Value.ValueKind == JsonValueKind.Null)
+                            if (property0.Value.ValueKind == JsonValueKind.Null || property0.Value.ValueKind == JsonValueKind.String && property0.Value.GetString().Length == 0)
```
Comment (annelo-msft) [inline sdk/cosmosdb/Azure.ResourceManager.CosmosDB/src/Generated/Models/CosmosDBAccountData.Serialization.cs:555]: Is this something that could be wrapped into a utility method?

Comment (live1206) [reply sdk/cosmosdb/Azure.ResourceManager.CosmosDB/src/Generated/Models/CosmosDBAccountData.Serialization.cs:555]: That would require change in autorest.csharp, it is like this everywhere else.

File: sdk/cosmosdb/Azure.ResourceManager.CosmosDB/CHANGELOG.md [hunk 3]
```diff
@@ -1,14 +1,9 @@
 # Release History
 
-## 1.4.0-beta.2 (Unreleased)
-
-### Features Added
-
-### Breaking Changes
+## 1.4.0-beta.2 (2023-05-22)
```
Comment (ArthurMa1978) [inline sdk/cosmosdb/Azure.ResourceManager.CosmosDB/CHANGELOG.md:3]: Can release with the coming refresh for model factory

**Review by ArthurMa1978 — APPROVED** [review]

File: sdk/cosmosdb/Azure.ResourceManager.CosmosDB/CHANGELOG.md (hunk 3 above)
Comment (live1206) [reply sdk/cosmosdb/Azure.ResourceManager.CosmosDB/CHANGELOG.md]: reset the release date

**Pull request merged**